source/stylist/fortran.py

```diff
@@ -295,3 +297,58 @@ def examine_fortran(self, subject: FortranSource) -> List[Issue]:
 
         issues.sort(key=lambda x: (x.filename, x.line, x.description))
         return issues
+
+
+class KindPattern(FortranRule):
+    def __init__(self,
+                 integer_pattern: Union[str, Pattern],
+                 real_pattern: Union[str, Pattern]):
+        self._patterns: Dict[str, Pattern] \
+            = defaultdict(lambda: re.compile(r'.*'))
+        if isinstance(integer_pattern, str):
+            self._patterns['integer'] = re.compile(integer_pattern)
+        else:
+            self._patterns['integer'] = integer_pattern
+        if isinstance(real_pattern, str):
+            self._patterns['real'] = re.compile(real_pattern)
+        else:
+            self._patterns['real'] = real_pattern
+
+    def examine_fortran(self, subject: FortranSource) -> List[Issue]:
+        issues: List[Issue] = []
+
+        candidates: List[Fortran2003.Base] = []
+        # Component variables
+        candidates.extend(
+            subject.find_all(Fortran2003.Data_Component_Def_Stmt))
+        # Component Procedure
+        candidates.extend(
+            subject.find_all(Fortran2003.Proc_Component_Def_Stmt))
+        # Procedure declaration
+        candidates.extend(
+            subject.find_all(Fortran2003.Procedure_Declaration_Stmt))
+        # Variable declaration
+        candidates.extend(
+            subject.find_all(Fortran2003.Type_Declaration_Stmt))
```

Comment:
Extremely minor so don't worry too much but defining then extending the list several times is possible slightly more wasteful than defining it with a big block (but I guess it's nicer / more readable this way, or is there some other reason I'm missing?)

Reply:
It's purely about being easy and logical to follow.